core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -97,30 +97,13 @@ oppia.factory('TopicUpdateService', [
       return _getParameterFromChangeDict(changeDict, 'new_value');
     };
 
-    /* For additionalStoryIds, canonicalStoryIds and uncategorizedSkillIds,
-    * the new_value is the updated list, so this function extracts the newly
-    * added id from the new_value list, compared to the old_value list.
-    */
-    var _getNewlyAddedIdFromChangeDict = function(changeDict) {
-      oldIdList = _getParameterFromChangeDict(changeDict, 'old_value');
-      newIdList = _getParameterFromChangeDict(changeDict, 'new_value');
-      newId = newIdList.filter(function(item){
-        return oldIdList.indexOf(item) === -1;
-      });
-      return newId[0];
-    };
-
-    /* For additionalStoryIds, canonicalStoryIds and uncategorizedSkillIds,
-    * the new_value is the updated list, so this function extracts the removed
-    * id from the old_value list, compared to the new_value list.
-    */
-    var _getNewlyRemovedIdFromChangeDict = function(changeDict) {
-      oldIdList = _getParameterFromChangeDict(changeDict, 'old_value');
-      newIdList = _getParameterFromChangeDict(changeDict, 'new_value');
-      newId = oldIdList.filter(function(item){
-        return newIdList.indexOf(item) === -1;
-      });
-      return newId[0];
+    var _checkIfSkillIdInList = function(skillId, skillList) {
+      for (var i = 0; i < skillList.length; i++) {
```

Comment:
Actually ... wouldn't `skillList.indexOf(skillId) !== -1` suffice?

also, I think skillList is misnamed. If its elements are skill IDs then it should be called skillIdList. In general, be very careful to distinguish between skill IDs and skills (or IDs and objects in general); they are different things. Make sure to double-check the variable names used in the callers, too.

Reply:
Yes, ok, done.